osu.Game/Overlays/MusicController.cs

```diff
@@ -166,10 +166,17 @@ public void EnsurePlayingSomething()
         /// <summary>
         /// Start playing the current track (if not already playing).
         /// </summary>
+        /// <param name="restart">Whether to restart the track from the beginning.</param>
+        /// <param name="requestedByUser">
+        /// Whether the request to play was issued by the user rather than internally.
+        /// Specifying <c>true</c> will ensure that other methods like <see cref="EnsurePlayingSomething"/>
+        /// will resume music playback going forward.
+        /// </param>
         /// <returns>Whether the operation was successful.</returns>
-        public bool Play(bool restart = false)
+        public bool Play(bool restart = false, bool requestedByUser = false)
         {
-            IsUserPaused = false;
+            if (requestedByUser)
+                IsUserPaused = false;
```

Comment:
This is very weird to me. So if you `Play(.., false)`, the music controlled can potentially be left in `IsUserPaused = true` but the track still playing?

Reply:
yep, for instance during gameplay. right now song select forcefully un(user)pauses on returning, but that may not be what we want at the end of the day, not sure.

Reply:
This variable needs to be renamed then. It can't remain as IsUser**Paused** when potentially it's not paused at all.

Needs to be UserPauseRequested, and appropriate documentation that says it doesn't necessarily mean the track is paused.

Reply:
no issues with that, go for it